eval.go

```diff
@@ -322,7 +323,7 @@ nextFileMatch:
 		visitMatches(mt, known, func(mt matchTree) {
 			atomMatchCount++
 		})
-		finalCands := gatherMatches(mt, known)
+		finalCands := gatherMatches(mt, known, !opts.ChunkMatches)
```

Comment:
callsite is a bit hard to read, had to jump to def. Maybe pull out `shouldMergeMatches := !opts.ChunkMatches`.

Maybe a suggestion due to the introduction of `merge` bool. Factor out the merge and sort code into there own functions. Then mutate finalCands. Maybe just over thinking it, bool is fine :)